File: LeetCode/0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.swift
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init() { self.val = 0; self.left = nil; self.right = nil; }
 *     public init(_ val: Int) { self.val = val; self.left = nil; self.right = nil; }
 *     public init(_ val: Int, _ left: TreeNode?, _ right: TreeNode?) {
 *         self.val = val
 *         self.left = left
 *         self.right = right
 *     }
 * }
 */
class Solution {
    func maxDepth(_ root: TreeNode?) -> Int {
        guard let root else { return 0 }
        
        var maxDepth = 0
        var queue = [(TreeNode, Int)]()
        queue.append((root, 1))
        
        while !queue.isEmpty {
            let (curNode, curDepth) = queue.removeFirst()
            maxDepth = max(curDepth, maxDepth)
            
            if let left = curNode.left {
                queue.append((left, curDepth + 1))
            }
            
            if let right = curNode.right {
                queue.append((right, curDepth + 1))
            }
        }
        
        return maxDepth
    }
}

File: LeetCode/0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        max_depth = 0
        if root is None:
            return max_depth
        
        queue = deque()
        queue.append((root, 1))

        while queue:
            (curr, height) = queue.popleft()
            max_depth = height
            if curr.left:
                queue.append((curr.left, height+1))
            if curr.right:
                queue.append((curr.right, height+1))
        
        return max_depth